app/js/components/manageApps/Addon.jsx

```diff
@@ -83,6 +90,71 @@ class Addon extends Component {
 
   }
 
+  fetchCurrentUser(){
+    const apiHelper = new ApiHelper(null);
+    const getUserData = new Promise(function(resolve, reject) {
+      apiHelper.get('/v1/session').then(response => {
+        resolve(response);
+      });
+    });
+    return getUserData;
+  }
+
+  checkIsAdmin() {
+    this.fetchCurrentUser().then((response) => {
+      if (response.user) {
+        const user = response.user;
+        let userPrivileges = [];
+        let userCanManageModules;
+        let userHasPermittedRoles;
+        userPrivileges = userPrivileges.concat(user.privileges);
+        if (user.roles.length > 0){
+          const roles = user.roles;
+          const userRoles = roles.find((userRole) => {
+            if (userRole.display === "System Developer" || userRole.display === "Provider" ) {
```

Comment:
There is no need to test for system developer or provider roles. All that one needs is the "Manage Modules" prvilege

Reply:
I tested for roles since the admin user we use for development do not have the list of priviledges so they are being locked out of the system and they cant view the add-ons manager that's why I resulted to using the roles as another way of giving access to developers so this also happens to the admin user on ref-app.  
@dkayiwa

Reply:
Since the user has the System Developer role, they by default assume that privilege. So you do not need to test for the system developer role

Reply:
I have fixed and stripped giving the Provider  the access